sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessorBuilder.java

```diff
@@ -149,6 +151,14 @@ public BatchSpanProcessorBuilder setMeterProvider(MeterProvider meterProvider) {
     return this;
   }
 
+  /** Sets the {@link InternalTelemetryVersion} defining which metrics this processor records. */
```

Comment:
I copied the pattern from https://github.com/open-telemetry/opentelemetry-java/pull/7265#discussion_r2069567357

I'm a bit confused though since it sounded like the intent was to "After some grace period remove support for the legacy metrics", but with this in the public API it seems not possible (I was expecting to use a system property for the switch). Guessing the ship has sailed but want to doublecheck

Reply:
Yes, looks like we have to support the legacy format forever.

I don't understand why we only have `LEGACY` and `LATEST` though - and not the `1_31` as proposed by @jack-berg initially.

Latest means you opt in to a breaking change if the semconv changes is the future, so only legacy is really stable.